_posts/2023-10-04-#28107.md

```diff
@@ -7,17 +7,55 @@ authors: [dergoegge]
 components: ["utils/log/libs"]
 host: stickies-v
 status: upcoming
-commit:
+commit: e74152d5b48bdca8dfc2217f35842dbb24b2746c
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- Since the activation of SegWit, transactions are identifiable by two types of identifiers: `txid` and `wtxid`. Both identifiers are a `uint256` resulting from a double `sha256` hash of serialized transaction data. The core [difference](https://github.com/bitcoin/bitcoin/blob/dcfbf3c2107c3cb9d343ebfa0eee78278dea8d66/src/primitives/transaction.h#L276-L280) is that `wtxid` includes the witness data, while `txid` does not.
+ 
+- Type-safety is a concept in programming to help avoid type errors, enforcing strict correspondence between a variable and its type. It ensures that a program performs only legal operations on data, with validation occurring during compile-time. This proactive error detection and prevention lead to more reliable and maintainable software, minimizing unintended behavior and vulnerabilities. An example of an earlier type-safety improvement in Bitcoin Core is the incorporation of `std::chrono`, introduced in [#16908](https://github.com/bitcoin/bitcoin/pull/16908).
+
+- In the context, of transaction identifiers, type-safety is important because it is impossible to deduce the type difference from just the `uint256` value. With `GenTxid` type checking can be done, but it is more verbose and done at run-time instead of at compile-time. Additionally, transaction identifier types improve readability of code by being explicit about expected types instead of solely relying on clear variable naming.
```

Comment:
```suggestion
- In the context, of transaction identifiers, type-safety is important because it is impossible to deduce the type difference from just the `uint256` value. With `GenTxid` type checking can be done, but it is more verbose and done at run-time instead of at compile-time. Additionally, transaction identifier types improve the readability of code by being explicit about expected types instead of solely relying on clear variable naming.
```